### ISSUE TITLE
`Add-NamespacePermission` fails with an error message - "Sequence contains no matching element and POST_VALIDATION_FAILED"

### ISSUE THREAD

**Issue by lksece** [vmware/power-validated-solutions-for-cloud-foundation#265]
### Code of Conduct

- [X] I have read and agree to the [Code of Conduct](https://github.com/vmware-samples/power-validated-solutions-for-cloud-foundation/blob/main/CODE_OF_CONDUCT.md).
- [ ] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [ ] Do not leave "+1" or other comments that do not add relevant information or questions.
- [ ] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### VMware Cloud Foundation

4.5.1 and 4.4.1

### Module Version

2.3.0.1005

### PowerShell Version

5.1.14393.5582

### PowerCLI Version

13.0.0 -revision 20829139

### PowerVCF Version

2.3.0.1004

### Guest Operating System

windows server 2016

### Description

Doc: https://docs.vmware.com/en/VMware-Cloud-Foundation/services/vcf-developer-ready-infrastructure-v1/GUID-B9210D55-B4CA-4623-8BAA-C91437DBA51A.html

```powershell
$sddcManagerFqdn = "sfo-vcf01.sfo.rainpole.io"
PS C:\Users\Administrator> $sddcManagerUser = "administrator@vsphere.local"
PS C:\Users\Administrator> $sddcManagerPass = "VMw@re123!"
PS C:\Users\Administrator> $sddcDomainName = "sfo-w01"
PS C:\Users\Administrator> $domainFqdn = "sfo.rainpole.io"
PS C:\Users\Administrator> $domainBindUser = "svc-vsphere-ad"
PS C:\Users\Administrator> $domainBindPass = "VMw@re123!"
PS C:\Users\Administrator> $wmNamespaceName = "sfo-w01-ns01"
PS C:\Users\Administrator> $wmNamespaceEditUserGroup = "gg-kub-admins"
PS C:\Users\Administrator> $wmNamespaceViewUserGroup = "gg-kub-readonly"
```

```powershell
 Add-NamespacePermission -server $sddcManagerFqdn -user $sddcManagerUser -pass $sddcManagerPas
s -sddcDomain $sddcDomainName -domain $domainFqdn -domainBindUser $domainBindUser -domainBindPass $domainBindPass -names
pace $wmNamespaceName -principal $wmNamespaceEditUserGroup -role edit -type group
New-WMNamespacePermission : 5/26/2023 4:47:36 PM        New-WMNamespacePermission               Sequence contains no matching element
At C:\Program Files\WindowsPowerShell\Modules\PowerValidatedSolutions\2.3.0.1005\PowerValidatedSolutions.psm1:6720 char:49
+ ...             New-WMNamespacePermission -Namespace $namespace -Role $ro ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [New-WMNamespacePermission], VimException
    + FullyQualifiedErrorId : Core_BaseCmdlet_UnknownError,VMware.VimAutomation.WorkloadManagement.Commands.Cmdlets.NewWMNamespacePermission

Assigning Role (edit) to group (gg-kub-admins) in Namespace (sfo-w01-ns01): SUCCESSFUL
PS C:\Users\Administrator> Add-NamespacePermission -server $sddcManagerFqdn -user $sddcManagerUser -pass $sddcManagerPass -sddcDomain $sddcDomainName -domain $domainFqdn -domainBindUser $domainBindUser -domainBindPass $domainBindPass
-namespace $wmNamespaceName -principal $wmNamespaceViewUserGroup -role view -type group
New-WMNamespacePermission : 5/26/2023 4:50:03 PM        New-WMNamespacePermission               Exception of type 'System.ArgumentOutOfRangeException' was thrown.
Parameter name: role
Actual value was VIEW.
At C:\Program Files\WindowsPowerShell\Modules\PowerValidatedSolutions\2.3.0.1005\PowerValidatedSolutions.psm1:6720 char:49
+ ...             New-WMNamespacePermission -Namespace $namespace -Role $ro ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [New-WMNamespacePermission], VimException
    + FullyQualifiedErrorId : Core_BaseCmdlet_UnknownError,VMware.VimAutomation.WorkloadManagement.Commands.Cmdlets.NewWMNamespacePermission

Get-WMNamespacePermission : 5/26/2023 4:50:03 PM        Get-WMNamespacePermission               Exception of type 'System.ArgumentOutOfRangeException' was thrown.
Parameter name: role
Actual value was VIEW.
At C:\Program Files\WindowsPowerShell\Modules\PowerValidatedSolutions\2.3.0.1005\PowerValidatedSolutions.psm1:6721 char:53
+ ...         if (Get-WMNamespacePermission -Namespace $namespace -Domain $ ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Get-WMNamespacePermission], VimException
    + FullyQualifiedErrorId : Core_BaseCmdlet_UnknownError,VMware.VimAutomation.WorkloadManagement.Commands.Cmdlets.GetWMNamespacePermission

Add-NamespacePermission : Assigning Role (view) to group (gg-kub-readonly) in Namespace (sfo-w01-ns01): POST_VALIDATION_FAILED
At line:1 char:1
+ Add-NamespacePermission -server $sddcManagerFqdn -user $sddcManagerUs ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Add-NamespacePermission
```

### Error or Debug Output

Assigning Role (view) to group (gg-kub-readonly) in Namespace (sfo-w01-ns01): POST_VALIDATION_FAILED

### Expected Behavior

User should be able assign the role without any error message.

### Actual Behavior

Though console shows the error message, it actually assigns role to the group.
![image](https://github.com/vmware-samples/power-validated-solutions-for-cloud-foundation/assets/28644973/5afadacc-5c53-4506-b1f8-336e2db4f7b5)

### Steps to Reproduce

Deploy VCF 441 or VCF 451
Configure DRI 
Run the above command as specified in the doc. https://docs.vmware.com/en/VMware-Cloud-Foundation/services/vcf-developer-ready-infrastructure-v1/GUID-B9210D55-B4CA-4623-8BAA-C91437DBA51A.html

**Comment by lksece** [comment]
same issue is observed for https://docs.vmware.com/en/VMware-Cloud-Foundation/services/vcf-developer-ready-infrastructure-v1/GUID-55E5A75E-6032-4484-A196-BACA4EB8CEED.html workflow as well.

**Comment by GaryJBlake** [comment]
Initial investigation looks like this is an issue with the Get-WMNamespacePermission cmdlet shipped as part of PowerCLI 13.0.0

**Comment by Ashwini-Ravi** [comment]
The issue is still present. Please find the screenshot attached
<img width="1240" alt="Screenshot 2023-07-19 at 3 37 13 PM" src="https://github.com/vmware-samples/power-validated-solutions-for-cloud-foundation/assets/122252705/9ca54efb-fef8-4a0d-a460-0c9e45ab8e29">
